src/framework/midi/midimodule.cpp

```diff
@@ -96,3 +102,19 @@ void MidiModule::onDeinit()
     m_midiOutPort->deinit();
     m_midiInPort->deinit();
 }
+
+void MidiModule::preamble(muse::audio::AudioModule* am)
+{
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
+    m_audioModule = std::shared_ptr<muse::audio::AudioModule>(am);
+#endif
+}
+
+mu::async::Channel<tick_t, mu::midi::Event>* MidiModule::getMidiInputQueue()
+{
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
+    return m_midiInPort->eventReceivedPtr();
```

Comment:
We don’t use these classes like that... they are not for providing data, they are just for initialization... There is no need to pass them to each other, and especially not to the driver initialization methods. 
To pass dependencies we use a "dependency injection system" 
For example, if you need this queue in some specific implementation class, then you can simply do: 

```
#include "modularity/ioc.h"
#include "midi/imidioutport.h"

.h 
class JackDriverState : public AudioDriverState, public async::Asyncable
{
    Inject<midi::IMidiOutPort> midiOutPort;
...

cpp

void JackDriverState::someMethof()
{
         mu::async::Channel<tick_t, mu::midi::Event> q = m_midiInPort->eventReceived();
         ...
}
```